libs/utils/src/tabulation/tally_reports.ts

```diff
@@ -0,0 +1,102 @@
+import { Candidate, CandidateContest, Tabulation } from '@votingworks/types';
+import { assertDefined } from '@votingworks/basics';
+import { combineCandidateContestResults } from './tabulation';
+
+type TallyReportCandidateRow = Candidate & {
+  scannedTally: number;
+  manualTally: number;
+};
+
+function isNonCandidateWriteInTally(
+  candidateTally: Tabulation.CandidateTally
+): boolean {
+  return (
+    candidateTally.id === Tabulation.PENDING_WRITE_IN_ID ||
+    candidateTally.id === Tabulation.GENERIC_WRITE_IN_ID
+  );
+}
+
+function getAllWriteInRows({
+  combinedContestResults,
+  scannedContestResults,
+  manualContestResults,
+}: {
+  combinedContestResults: Tabulation.CandidateContestResults;
+  scannedContestResults: Tabulation.CandidateContestResults;
+  manualContestResults?: Tabulation.CandidateContestResults;
+}): TallyReportCandidateRow[] {
+  const rows: TallyReportCandidateRow[] = [];
+  const writeInCandidateTallies: Tabulation.CandidateTally[] = [];
+  const otherWriteInTallies: Tabulation.CandidateTally[] = [];
+
+  for (const candidateTally of Object.values(combinedContestResults.tallies)) {
+    if (candidateTally.isWriteIn) {
+      if (isNonCandidateWriteInTally(candidateTally)) {
+        otherWriteInTallies.push(candidateTally);
+      } else {
+        writeInCandidateTallies.push(candidateTally);
+      }
+    }
+  }
+
+  // list write-in candidates first, then other write-in counts
+  for (const candidateTally of [
+    ...writeInCandidateTallies,
+    ...otherWriteInTallies,
+  ]) {
+    rows.push({
+      ...candidateTally,
+      scannedTally:
+        scannedContestResults.tallies[candidateTally.id]?.tally ?? 0,
+      manualTally: manualContestResults?.tallies[candidateTally.id]?.tally ?? 0,
+    });
+  }
+
+  return rows;
+}
+
+export function getTallyReportCandidateRows({
+  contest,
+  scannedContestResults,
+  manualContestResults,
+}: {
+  contest: CandidateContest;
+  scannedContestResults: Tabulation.CandidateContestResults;
+  manualContestResults?: Tabulation.CandidateContestResults;
+}): TallyReportCandidateRow[] {
```

Comment:
I think that for the use case in this PR, this is a little overbuilt. But. I'm breaking this logic out into a util because I will use the same logic for frontend tally reports, and add different algorithms for grouping / showing write-in candidates. Either "show all",  or "show significant." Future features, like the ability to have official write-in candidates, could add more algorithms here. Will build on this in next PR.